lib/df-detection.c

```diff
@@ -7,7 +7,7 @@
 #include <linux/slab.h>
 #include <linux/stacktrace.h>
 
-void add_address(const void *addr, unsigned long len)
+void add_address(const void *addr, unsigned long len,unsigned int caller)
```

Comment:
Nit: space after comma.

Reply:
Also, `caller` must be unsigned long (see definition of `_RET_IP_`: https://elixir.bootlin.com/linux/latest/source/include/linux/kernel.h#L170)
If you pass it as unsigned int, the top bytes will be truncated, which may result in an incorrect address being stored/printed.

Reply:
And you need to actually store `caller` in `current->addresses`

Reply:
okay i will change it to unsigned long sorry for that.
i already saved the caller in addresses struct